tests/accessor/accessor_common.h

```diff
@@ -31,6 +33,107 @@ enum class accessor_type {
   local_accessor,
   host_accessor,
 };
+}  // namespace accessor_tests_common
+
+namespace Catch {
+template <>
+struct StringMaker<accessor_tests_common::accessor_type> {
+  using type = accessor_tests_common::accessor_type;
+  static std::string convert(type value) {
+    switch (value) {
+      case type::generic_accessor:
+        return "sycl::accessor";
+      case type::local_accessor:
+        return "sycl::local_accessor";
+      case type::host_accessor:
+        return "sycl::host_accessor";
+      default:
+        return "unknown accessor type";
+    }
+  }
+};
+
+template <>
+struct StringMaker<sycl::access_mode> {
+  using type = sycl::access_mode;
+  static std::string convert(type value) {
+    switch (value) {
+      case type::read:
+        return "access_mode::read";
+      case type::write:
+        return "access_mode::write";
+      case type::read_write:
+        return "access_mode::read_write";
+      default:
+        // no stringification for deprecated ones
+        return "unknown access mode";
+    }
+  }
+};
+
+template <>
+struct StringMaker<sycl::target> {
+  using type = sycl::target;
+  static std::string convert(type value) {
+    switch (value) {
+      case type::device:
+        return "target::device";
+      case type::host_task:
+        return "target::host_task";
+      default:
+        // no stringification for deprecated ones
+        return "unknown target";
+    }
+  }
+};
+}  // namespace Catch
+
+namespace accessor_tests_common {
+/**
+ * @brief Builder for a section name with the fluent interface
+ * @details Be aware that Catch2 doesn't support nested sections with the same
+ *          name, see https://github.com/catchorg/Catch2/issues/816 for details.
+ *          So if you see
+ *              Assertion `m_parent' failed.
+ *          that's probably the case.
+ */
+class section_name {
+  std::string m_description;
+  std::ostringstream m_parameters;
+
+ public:
+  section_name(const section_name& other)
+      : m_description(other.m_description),
+        m_parameters(other.m_parameters.str()) {}
+
+  // Avoid implicit move constructor removal
+  section_name(section_name&& other) = default;
+
+  section_name(const std::string& description) : m_description(description) {
+    if (m_description.empty()) {
+      m_description = "run";
+    }
+  }
+
+  template <typename T>
+  section_name& with(const std::string& name, T&& value) {
+    m_parameters << ' ' << name << ": "
+                 << Catch::StringMaker<T>::convert(std::forward<T>(value))
+                 << ',';
+    return *this;
+  }
+
+  std::string create() const {
+    std::string result("Test " + m_description);
```

Comment:
I understand that "Test" came from previous version of get_section_name but do we really need that? Most section descriptions in accessor tests right now start description as sentence.